services/src/machine_learning/xgb_config_builder.rs

```diff
@@ -0,0 +1,1057 @@
+#[derive(Builder, Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[builder(default)]
+pub struct LinearBoosterParameters {
+    lambda: f32,
+    alpha: f32,
+    updater: LinearUpdate,
+}
+
+impl LinearBoosterParameters {
+    pub fn as_string_pairs(&self) -> Vec<(String, String)> {
+        let v = vec![
+            ("booster".to_owned(), "gblinear".to_owned()),
+            ("lambda".to_owned(), self.lambda.to_string()),
+            ("alpha".to_owned(), self.alpha.to_string()),
+            ("updater".to_owned(), self.updater.to_string()),
+        ];
+
+        v
+    }
+}
+
+impl Default for LinearBoosterParameters {
+    fn default() -> Self {
+        LinearBoosterParameters {
+            lambda: 0.0,
+            alpha: 0.0,
+            updater: LinearUpdate::default(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub enum TreeMethod {
+    #[default]
+    Auto,
+    Exact,
+    Approx,
+    Hist,
+    GpuExact,
+    GpuHist,
+}
+
+impl ToString for TreeMethod {
+    fn to_string(&self) -> String {
+        match *self {
+            TreeMethod::Auto => "auto".to_owned(),
+            TreeMethod::Exact => "exact".to_owned(),
+            TreeMethod::Approx => "approx".to_owned(),
+            TreeMethod::Hist => "hist".to_owned(),
+            TreeMethod::GpuExact => "gpu_exact".to_owned(),
+            TreeMethod::GpuHist => "gpu_hist".to_owned(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum TreeUpdater {
+    GrowColMaker,
+    DistCol,
+    GrowHistMaker,
+    GrowLocalHistMaker,
+    GrowSkMaker,
+    Sync,
+    Refresh,
+    Prune,
+}
+
+impl ToString for TreeUpdater {
+    fn to_string(&self) -> String {
+        match *self {
+            TreeUpdater::GrowColMaker => "grow_colmaker".to_owned(),
+            TreeUpdater::DistCol => "distcol".to_owned(),
+            TreeUpdater::GrowHistMaker => "grow_histmaker".to_owned(),
+            TreeUpdater::GrowLocalHistMaker => "grow_local_histmaker".to_owned(),
+            TreeUpdater::GrowSkMaker => "grow_skmaker".to_owned(),
+            TreeUpdater::Sync => "sync".to_owned(),
+            TreeUpdater::Refresh => "refresh".to_owned(),
+            TreeUpdater::Prune => "prune".to_owned(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub enum ProcessType {
+    #[default]
+    Default,
+    Update,
+}
+
+impl ToString for ProcessType {
+    fn to_string(&self) -> String {
+        match *self {
+            ProcessType::Default => "default".to_owned(),
+            ProcessType::Update => "update".to_owned(),
+        }
+    }
+}
+
+impl<'a> From<&'a str> for ProcessType {
+    fn from(s: &'a str) -> Self {
+        match s {
+            "default" => ProcessType::Default,
+            "update" => ProcessType::Update,
+            // TODO: better error handling
+            _ => panic!("no known process_type for {s}"),
```

Comment:
Implement TryFrom instead of From and return an error

Reply:
see: https://github.com/geo-engine/geoengine/pull/746#discussion_r1252302470